### PR TITLE
[expo-notifications] Update NCL example

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Fixes the notifications screen in the unversioned NCL. 
Updates the example - use the new notifications API. 

# Test Plan

- NCL:
  - Android ✅
  - iOS ✅

### CODE REVIEW

File: apps/native-component-list/src/screens/Location/GeofencingScreen.tsx [hunk 1]
```diff
@@ -189,10 +186,13 @@ TaskManager.defineTask(GEOFENCING_TASK, async ({ data: { region } }: { data: any
...
-    title: 'Expo Geofencing',
-    body: `You're ${stateString} a region ${region.identifier}`,
-    data: region,
+  await Notifications.scheduleNotificationAsync({
+    content: {
+      title: 'Expo Geofencing',
+      body: `You're ${stateString} a region ${region.identifier}`,
+      data: region,
+    },
+    trigger: null,
```
Comment (tsapeta) [inline apps/native-component-list/src/screens/Location/GeofencingScreen.tsx:195]: Is it required to explicitly pass `null` as a trigger?

**Review by tsapeta — APPROVED** [review]
Looks good, thanks for updating it 👍

File: apps/native-component-list/src/screens/Location/GeofencingScreen.tsx (hunk 1 above)
Comment (lukmccall) [reply apps/native-component-list/src/screens/Location/GeofencingScreen.tsx:195]: cc @sjchmiela

Comment (lukmccall) [reply apps/native-component-list/src/screens/Location/GeofencingScreen.tsx:195]: I've copied it from the documentation. So, I assumed that it's necessary to pass `null`.  But, I'm not sure now.

Comment (lukmccall) [reply apps/native-component-list/src/screens/Location/GeofencingScreen.tsx:195]: It's necessary 😅 
image

Comment (tsapeta) [reply apps/native-component-list/src/screens/Location/GeofencingScreen.tsx:195]: UX designers are crying 😢

**Review by esamelson — APPROVED** [review]

**Pull request merged**